pycbc/inference/sampler_emcee.py

```diff
@@ -0,0 +1,993 @@
+            `WaveformArray` (as long as the file contains the necessary fields
+            to derive the virtual field or method), and/or a function of
+            these.
+        thin_start : int
+            Index of the sample to begin returning samples. Default is to read
+            samples after burn in. To start from the beginning set thin_start
+            to 0.
+        thin_interval : int
+            Interval to accept every i-th sample. Default is to use the
+            `fp.acl`. If `fp.acl` is not set, then use all samples
+            (set thin_interval to 1).
+        thin_end : int
+            Index of the last sample to read. If not given then
+            `fp.niterations` is used.
+        iteration : int
+            Get a single iteration. If provided, will override the
+            `thin_{start/interval/end}` arguments.
+        walkers : {None, (list of) int}
+            The walker index (or a list of indices) to retrieve. If None,
+            samples from all walkers will be obtained.
+        temps : {None, (list of) int, 'all'}
+            The temperature index (or list of indices) to retrieve. If None,
+            only samples from the coldest (= 0) temperature chain will be
+            retrieved. To retrieve all temperates pass 'all', or a list of
+            all of the temperatures.
+        flatten : {True, bool}
+            The returned array will be one dimensional, with all desired
+            samples from all desired walkers concatenated together. If False,
+            the returned array will have dimension requested temps x requested 
+            walkers x requested iterations.
+        samples_group : {None, str}
+            The group in `fp` from which to retrieve the parameter fields. If
+            None, searches in `fp.samples_group`.
+        array_class : {None, array class}
+            The type of array to return. The class must have a `from_kwargs`
+            class method and a `parse_parameters` method. If None, will return
+            a WaveformArray.
+
+        Returns
+        -------
+        array_class
+            Samples for the given parameters, as an instance of a the given
+            `array_class` (`WaveformArray` if `array_class` is None).
+        """
+        # get the group to load from
+        if samples_group is None:
+            samples_group = fp.samples_group
+        # get the type of array class to use
+        if array_class is None:
+            array_class = WaveformArray
+        # get the names of fields needed for the given parameters
+        possible_fields = dict([[str(name), float]
+            for name in fp[fp.samples_group].keys()])
+        loadfields = array_class.parse_parameters(parameters,
+            possible_fields=possible_fields)
+        return cls._read_fields(fp, samples_group, loadfields, array_class,
+                thin_start=thin_start, thin_interval=thin_interval,
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->